src/sync/utils/create-build-array-actions.js

```diff
@@ -0,0 +1,107 @@
+const REGEX_NUMBER = new RegExp(/^\d+$/)
+const REGEX_UNDERSCORE_NUMBER = new RegExp(/^_\d+$/)
+
+export const ADD_ACTIONS = 'create'
+export const REMOVE_ACTIONS = 'remove'
+export const CHANGE_ACTIONS = 'change'
+
+/**
+ * Generate + configure a function to build actions for nested objects
+ * @param  {string} key    key of the attribute containing the array of
+ *   nested objects
+ * @param  {object} config configuration object that can contain the keys
+ *   [ADD_ACTIONS, REMOVE_ACTIONS, CHANGE_ACTIONS], each of
+ *   which is a function. The function should accept the old + new arrays and
+ *   return an action object.
+ * @return {Array}        The generated array of actions
+ */
+export default function createBuildArrayActions (key, config) {
+  return function buildNestedObjectActions (diff, oldObj, newObj) {
+    const addActions = []
+    const removeActions = []
+    const changeActions = []
+
+    if (diff[key]) {
+      const nestedObjects = diff[key]
+      Object.keys(nestedObjects).forEach((index) => {
+        if (config[ADD_ACTIONS] && isCreateAction(nestedObjects, index))
+          addActions.push(
+            config[ADD_ACTIONS](oldObj[key], newObj[key], index)
```

Comment:
@PhilippSpo suggested to actually pass the "oldValue" and "newValue" directly, that way we don't need to pass the index anymore. This covers all current cases already.

We need to differentiate between the different cases though:

```js
// When adding a new element you don't need the oldObj
if (config[ADD_ACTIONS] && isCreateAction(nestedObjects, index))
  addActions.push(newObj[key][index])

// When removing an existing element you don't need the newObj
if (config[REMOVE_ACTIONS] && isRemoveAction(nestedObjects, index)) {
  const realIndex = index.replace('_', '')
  removeActions.push(oldObj[key][realIndex])
}

// When changing an existing element you need both
if (config[CHANGE_ACTIONS] && isChangeAction(nestedObjects, index)) {
  const oldValue = oldObj[key][index]
  const newValue = newObj[key][index]
  changeActions.push(oldValue, newValue)
}
```